go/vt/mysqlctl/query.go

```diff
@@ -213,16 +213,28 @@ func (mysqld *Mysqld) fetchVariables(ctx context.Context, pattern string) (map[s
 const (
 	masterPasswordStart = "  MASTER_PASSWORD = '"
 	masterPasswordEnd   = "',\n"
+	passwordStart       = " PASSWORD = '"
+	passwordEnd         = "'"
 )
 
-func redactMasterPassword(input string) string {
+func redactPassword(input string) string {
 	i := strings.Index(input, masterPasswordStart)
+	// We have master password in the query, try to redact it
+	if i != -1 {
+		j := strings.Index(input[i+len(masterPasswordStart):], masterPasswordEnd)
+		if j == -1 {
+			return input
+		}
+		input = input[:i+len(masterPasswordStart)] + strings.Repeat("*", j) + input[i+len(masterPasswordStart)+j:]
```

Comment:
Why are we preserving the lenght of the password, should we not keep it a constant length? We do not want people to be able to infer the length either from the logs.
Otherwise, LGTM

Reply:
Yeah, good point - updated